website/docs/02-separation-logic-and-biabduction.md

```diff
@@ -0,0 +1,264 @@
+---
+
+export const Math = ({ code }) => (
+<img
+src={`https://math.now.sh?from=${code}&color=mediumslateblue`}
+style={{
+      height: "100%",
+      verticalAlign: "middle"
+    }}
+/>
+);
+
+- [Separation logic](docs/separation-logic-and-bi-abduction.html#separation-logic)
+- [Bi-abduction](docs/separation-logic-and-bi-abduction.html#bi-abduction)
+- [Technical papers](docs/separation-logic-and-bi-abduction.html#technical-papers)
+
+## Separation logic
+
+Separation logic is a novel kind of mathematical logic which facilitates reasoning about
+mutations to computer memory. It enables scalability by breaking reasoning into chunks
+corresponding to local operations on memory, and then composing
+the reasoning chunks together.
+
+Separation logic is based on a logical connective <Math code="*" /> called the _separating conjunction_ and pronounced "and separately". Separation logic formulae are interpreted over program allocated heaps. The logical formula
+<Math code="A*B" /> holds of a piece of program heap (a heaplet) when it can be divided into two sub-heaplets described by <Math code="A" /> and <Math code="B" />.
+For example, the formula
+
+---
+
+<Math code="x%20%5Cmapsto%20y%20*%20y%20%5Cmapsto%20x" />
+
+---
+
+can be read "<Math code="x" /> points to <Math code="y" /> and separately <Math code="y" /> points to <Math code="x" />". This formula describes precisely two allocated memory cells. The first cell is allocated at the address denoted by the pointer <Math code="x" /> and the content of this cell is the value of <Math code="y" />.
+The second cell is allocated at the address denoted by the pointer <Math code="y" /> and the content of this second cell is the value of <Math code="x" />. Crucially, we know that there are precisely two cells because <Math code="*" /> stipulates that they are separated and therefore the cells are allocated in two different parts of memory. In other words, <Math code="*" />
+says that <Math code="x" /> and <Math code="y" /> do not hold the same value (i.e., these pointers are not aliased).
+The heaplet partitioning defined by the formula above can visualized like so:
+
+![](/img/SepSplit.jpg)
+
+The important thing about separating conjunction is
+the way that it fits together with mutation to computer memory; reasoning about program commands
+tends to work by updating <Math code="*" />-conjuncts in-place, mimicking the operational in-place update of RAM.
+
+Separation logic uses Hoare triples of the form <Math code="%5Clbrace%20pre%20%5Crbrace%20prog%20%5Clbrace%20post%20%5Crbrace" /> where <Math code="pre" /> is the precondition, <Math code="prog" /> a program part, and <Math code="post" />
+the postcondition. Triples are abstract specifications of the behavior of the program. For example, we could take
+
+---
+
+<Math code="%5Clbrace%20r%20%5Cmapsto%20open%5Crbrace%20%5C%2C%20closeResource(r)%5C%2C%20%5Clbrace%20r%20%5Cmapsto%20closed%5Crbrace%20%20%5C%3B%5C%3B%5C%3B%20(spec)" />
```

Comment:
Did you generate the code in this page using another tool? What's the workflow for writing formulas like this? We are much more familiar with LaTeX, it was nice to be able to write LaTeX and get it rendered.

Reply:
Indeed. I added support for writing LaTeX directly!

Reply:
I don't understand: previously this was written as latex. Now it is written as something else with lots of escape characters. Where did the latex go?

Reply:
I haven't pushed latest changes to this branch yet. We can write latex like this in latest version.
![image](https://user-images.githubusercontent.com/38746192/70070038-3915da80-1636-11ea-9805-6ebb7391edc8.png)

Reply:
Ahh, beautiful, thanks :)

Reply:
👍